packages/components/src/internal/components/entities/models.ts

```diff
@@ -100,7 +100,7 @@ export class EntityParentType extends Record({
 
     generateFieldKey(): string {
         const parentInputType = this.getInputType();
-        const formattedQueryName = capitalizeFirstChar(this.query);
+        const formattedQueryName = this.label ?? capitalizeFirstChar(this.query);
```

Comment:
This doesn't seem correct. I would not expect the label to be "weight bearing" in terms of the underlying column that is generated. This relies on external usages of the model to understand that the label is leveraged in this way and that it must match the query name. What is it about the usages that are not handing in the correct "query"? Can we fix those and/or introduce an additional prop that is more explicit about what is going on here?

Reply:
The only difference between the EntityParentType label and query props is casing. The query is lowercase so when used in this generateFieldKey() helper it gets capitalized resulting in a fieldKey like "MaterialInputs/Tissue samples". The change here is just so that the fielKey generated would be "MaterialInputs/Tissue Samples". This just makes it easier for the test code to find these columns in the Editable Grid for sample insert since it can just use the expected query/table labels instead of having to do "toLowerCase() and capitalize()".

Reply:
thinking about this a bit more, maybe we should drop the "label" prop here all together and just stop doing the toLowerCase() on the query name so early (i.e. put the query name into the model and then usages of that "query" prop can toLowerCase() if they need to. I can look into that, likely a bit larger change but shouldn't be too bad